api.go

```diff
@@ -33,37 +33,39 @@ func (a *API) HandleDirectory(dir string) error {
 		return fmt.Errorf("Cannot list clang-c directory: %v", err)
 	}
 
-	h := &HeaderFile{
-		api: a,
-
-		dir: dir,
-
-		lookupEnum:        map[string]*Enum{},
-		lookupNonTypedefs: map[string]string{},
-		lookupStruct: map[string]*Struct{
-			"cxstring": &Struct{
-				Name:  "cxstring",
-				CName: "CXString",
-			},
-		},
-	}
+	var headerFiles []*HeaderFile
 
 	for _, hf := range headers {
-		name := hf.Name()
-
-		if hf.IsDir() || !strings.HasSuffix(name, ".h") {
+		if hf.IsDir() || !strings.HasSuffix(hf.Name(), ".h") {
 			continue
 		}
 
-		h.name = dir + name
+		h := &HeaderFile{
+			api: a,
+
+			dir:  dir,
+			name: dir + hf.Name(),
+
+			lookupEnum:        map[string]*Enum{},
+			lookupNonTypedefs: map[string]string{},
+			lookupStruct: map[string]*Struct{
+				"cxstring": &Struct{
+					Name:  "cxstring",
+					CName: "CXString",
+				},
+			},
+		}
+		headerFiles = append(headerFiles, h)
```

Comment:
Since the HeaderFile "h" is not ready yet, it should not be added right away. Please move this append at the end of the block.